examples_tests/61.UI/UI_System.cpp

```diff
@@ -0,0 +1,1102 @@
+		//io.Fonts->AddFontFromFileTTF("../../misc/fonts/Cousine-Regular.ttf", 15.0f);
+		//io.Fonts->AddFontFromFileTTF("../../misc/fonts/DroidSans.ttf", 16.0f);
+		//io.Fonts->AddFontFromFileTTF("../../misc/fonts/ProggyTiny.ttf", 10.0f);
+		//ImFont* font = io.Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\ArialUni.ttf", 18.0f, NULL, io.Fonts->GetGlyphRangesJapanese());
+		ImGuiIO& io = ImGui::GetIO();
+
+		uint8_t* pixels = nullptr;
+		int32_t width, height;
+		io.Fonts->GetTexDataAsRGBA32(&pixels, &width, &height);
+		assert(pixels != nullptr);
+		assert(width > 0);
+		assert(height > 0);
+		size_t const componentsCount = 4;
+		size_t const image_size = width * height * componentsCount * sizeof(uint8_t);
+
+		// Pixels buffer-----------------------
+		smart_refctd_ptr<ICPUBuffer> imageCpuBuffer = make_smart_refctd_ptr<ICPUBuffer>(image_size);
+		memcpy(imageCpuBuffer->getPointer(), pixels, imageCpuBuffer->getSize());   // TODO: Can we avoid this copy ?
+
+		// Cpu image---------------------------
+		ICPUImage::SCreationParams creationParams{};
+		creationParams.flags = static_cast<IImage::E_CREATE_FLAGS>(0u);
+		creationParams.type = IImage::ET_2D;
+		creationParams.format = EF_R8G8B8A8_UNORM;
+		creationParams.extent = { static_cast<uint32_t>(width), static_cast<uint32_t>(height), 1u };
+		creationParams.mipLevels = 1;
+		creationParams.arrayLayers = 1u;
+		creationParams.samples = IImage::ESCF_1_BIT;
+		creationParams.tiling = IImage::ET_OPTIMAL;
+
+		// TODO: Check usage
+		creationParams.usage = bitflag(IImage::EUF_SAMPLED_BIT) | IImage::EUF_TRANSFER_DST_BIT;
+		creationParams.sharingMode = ESM_EXCLUSIVE;
+		creationParams.queueFamilyIndexCount = 1u;
+		creationParams.queueFamilyIndices = nullptr;
+		creationParams.initialLayout = EIL_UNDEFINED;
+
+		auto const imageRegions = make_refctd_dynamic_array<smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1ull);
+		imageRegions->begin()->bufferOffset = 0ull;
+		imageRegions->begin()->bufferRowLength = creationParams.extent.width;
+		imageRegions->begin()->bufferImageHeight = 0u;
+		imageRegions->begin()->imageSubresource = {};
+		imageRegions->begin()->imageSubresource.aspectMask = IImage::EAF_COLOR_BIT;
+		imageRegions->begin()->imageSubresource.layerCount = 1u;
+		imageRegions->begin()->imageOffset = { 0, 0, 0 };
+		imageRegions->begin()->imageExtent = { creationParams.extent.width, creationParams.extent.height, 1u };
+
+		auto const cpuImage = ICPUImage::create(std::move(creationParams));
+		cpuImage->setBufferAndRegions(std::move(imageCpuBuffer), imageRegions);
+
+		// Gpu image------------------------------
+
+		// Creating gpu image then copying data into it which needs a transient command buffer
+		cpu2GpuParams.beginCommandBuffers();
+		IGPUObjectFromAssetConverter cpu2gpuConverter{};
+		auto const gpuImage = cpu2gpuConverter.getGPUObjectsFromAssets(
+			&cpuImage,
+			&cpuImage + 1,
+			cpu2GpuParams
+		);
+		cpu2GpuParams.waitForCreationToComplete(false);
```

Comment:
its a decent enough approach, but you should proably create a IGPUImage straight away and then use the video::IUTilities class to "copy buffer to image"

I guess fix it last.

Reply:
@deprilula28 you might want to fix this in your fork